src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java

```diff
@@ -444,9 +448,13 @@ MethodHandle viewAsType(MethodType newType, boolean strict) {
         @Override
         Object checkReceiver(Object recv) {
             if (!refc.isInstance(recv)) {
-                String msg = String.format("Receiver class %s does not implement the requested interface %s",
-                                           recv.getClass().getName(), refc.getName());
-                throw new IncompatibleClassChangeError(msg);
+                if (recv != null) {
+                    String msg = String.format("Receiver class %s does not implement the requested interface %s",
+                                               recv.getClass().getName(), refc.getName());
+                    throw new IncompatibleClassChangeError(msg);
+                } else {
+                    throw new NullPointerException();
```

Comment:
Since we know which `member` this is trying to invoke, the **NPE** can be constructed with a [JEP 358](https://openjdk.java.net/jeps/358)‑style message.